src/Qir/Runtime/stdlib/include/qir_stdlib.h

```diff
@@ -0,0 +1,230 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#include <cstdint>
+
+#ifdef _WIN32
+#define QIR_SHARED_API __declspec(dllexport)
+#else
+#define QIR_SHARED_API
+#endif
+
+#ifdef __cplusplus
+extern "C"
+{
+#endif
+
+    struct QirString;
+    struct QirArray;
+    struct QirCallable;
+    struct QirTuple;
+    struct QirBigInt;
+
+    enum PauliId : int8_t
+    {
+        PauliId_I = 0,
+        PauliId_X = 1,
+        PauliId_Z = 2,
+        PauliId_Y = 3,
+    };
+
+    typedef void (*t_CallableEntry)(QirTuple*, QirTuple*, QirTuple*);
+    typedef void (*t_CaptureCallback)(QirTuple*, int32_t);
+
+    // Returns a pointer to the malloc-allocated block.
+    QIR_SHARED_API char* __quantum__rt__memory_allocate(uint64_t size); // NOLINT
```

Comment:
Nit, FYI:
I'm not sure why this has not been done earlier but it makes more sense for this function to return `void*` rather than `char*`.
```suggestion
QIR_SHARED_API void* __quantum__rt__memory_allocate(uint64_t size); // NOLINT
```